File: app/src/main/java/com/sanislo/movieapp/domain/upcoming/UpcomingMoviesRepository.java
package com.sanislo.movieapp.domain.upcoming;

import android.arch.lifecycle.LiveData;
import android.arch.paging.PagedList;

import com.sanislo.movieapp.domain.model.MovieListItemModel;
import com.sanislo.movieapp.persistence.entity.MovieListItemEntity;
import com.sanislo.movieapp.persistence.response.movieList.MovieListResponse;

import io.reactivex.Single;

public interface UpcomingMoviesRepository {
    Single<MovieListResponse> getUpcoming(int page);
    Single<MovieListResponse> getUpcoming();
    void saveUpcomingMovies(MovieListResponse movieListResponse);
    LiveData<PagedList<MovieListItemModel>> getUpcomingLiveData();
}

File: app/src/main/java/com/sanislo/movieapp/domain/upcoming/UpcomingMoviesRepository.kt
package com.sanislo.movieapp.domain.upcoming

import androidx.lifecycle.LiveData
import androidx.paging.PagedList
import com.sanislo.movieapp.domain.model.MovieListItemModel
import com.sanislo.movieapp.persistence.response.movieList.MovieListResponse
import io.reactivex.Single

interface UpcomingMoviesRepository {
    fun upcoming(): Single<MovieListResponse>
    fun upcomingMoviesLiveData(): LiveData<PagedList<MovieListItemModel>>
    fun getUpcoming(page: Int): Single<MovieListResponse>
    fun refreshUpcoming(): Single<MovieListResponse>
    fun saveUpcomingMovies(movieListResponse: MovieListResponse)
}
